### PR TITLE
ImportAttributes should go through the same emit phases when in an ImportTypeNode

### LINKED ISSUE #56394: Inferring return types with import attributes causes debug assertion.
### 🔎 Search Terms

infer return type import attributes

### 🕗 Version & Regression Information

- This is a crash

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/dev/bug-workbench/?module=199&traceResolution=true#code/PTAEAEFsHsBMFcA2BTAXKAcnZHkA8AXAWAChSQIAzASxQDsBDSNUO7AfRgRQGdhq6dZACd+dWPgB0sSZAI9S+AA7RhBUAGNodHuuQ9IoALygCw+MgDcpcmHA16TFmwmc4SfWKGilDDQGsGAHNkSQArHm1SAG9QACJGZjj0OIFvOIAaeOVVeWTQWLjJYEgI-KKxCTxZMtAAX3qbEgp7WmRElkpoaEl5RTwVNVBgACoGEeBKeDoNAmptUYAjCdBKOgAKAEoC0lA90GFkAnhhOlBqSEGCddTBERKyzdI6oA)

### 💻 Code

```ts
// @module: NodeNext

// @filename: node_modules/inner/index.d.mts
export const esm = true;

// @filename: node_modules/inner/package.json
{ "name": "inner", "exports": { "./mjs": "./index.mjs" } }

// @filename: foo.ts
export /*a*/function/*b*/ fn() {
    return import("inner/mjs")
}
```

### 🙁 Actual behavior

Using infer return type on `fn` will cause a debug assertion. 

### 🙂 Expected behavior

Should infer the return type.

### Additional information about the issue

Found while working on isolated declaration.

It's caused by `ImportAttributes` not going through the full emit pipeline and thus not having their position assigned if they are part of an import type node.

### PR DESCRIPTION
**Pull request by dragomirtitian**
Fixes #56394

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -4077,6 +4071,18 @@ export function createPrinter(printerOptions: PrinterOptions = {}, handlers: Pri
         writeTrailingSemicolon();
     }
 
+    function emitTypeImportAttributes(node: ImportAttributes) {
```
Comment (jakebailey) [inline src/compiler/emitter.ts:4074]: I feel like this function name (and new enum value name) may be misleading; they don't have anything to do with type imports. Maybe this should be `emitImportTypeNodeAttributes`?

Comment (dragomirtitian) [reply src/compiler/emitter.ts:4074]: Renamed.

**Review by jakebailey — APPROVED** [review]

File: src/compiler/types.ts [hunk 2]
```diff
@@ -8141,13 +8141,14 @@ export const enum ExternalEmitHelpers {
...
-    EmbeddedStatement,   // Emitting an embedded statement
-    JsxAttributeValue,   // Emitting a JSX attribute value
+    SourceFile,              // Emitting a SourceFile
+    Expression,              // Emitting an Expression
+    IdentifierName,          // Emitting an IdentifierName
+    MappedTypeParameter,     // Emitting a TypeParameterDeclaration inside of a MappedTypeNode
+    Unspecified,             // Emitting an otherwise unspecified node
+    EmbeddedStatement,       // Emitting an embedded statement
+    JsxAttributeValue,       // Emitting a JSX attribute value
+    ImportTypeNodeAttributes,// Emitting attributes as part of an ImportTypeNode
```
Comment (jakebailey) [inline src/compiler/types.ts:8151]: Not sure how much this matters given we seem to be adding stuff here, I do wonder if new things should be marked as internal. @rbuckton probably knows better than me for this bit.

**Comment by dragomirtitian** [comment]
@jakebailey This one would also ideally make it in before isolated declarations. It causes a failure in a test. Not critical but nice to have.

File: src/compiler/types.ts (hunk 2 above)
Comment (rbuckton) [reply src/compiler/types.ts:8151]: I think it's fine for this to be public, as it is observable to custom transformers.

**Review by rbuckton — APPROVED** [review]

**Pull request merged**